src/frontend.md

```diff
@@ -5,31 +5,56 @@
 ```
 src/
   components/ (can be dumb and smart)
-    bankAccount/
+    [All components should be placed in some entity folder or shared folder]
+    bankAccount/ (entity, module, .etc)
       BankAccountTable/
         [no nested directories]
+        [avoid index.ts]
         BankAccountTable.ts
         BankAccountTable.test.ts
         BankAccountTableBadge.ts (child component)
         BankAccountTable.styled.ts
+        BankAccountTable.hook.ts (ui hook used only in the current component)
+        icon.svg (the icon used only in the current component)
+        [avoid index.ts]
+        [other files should be placed in queries or services]
     shared/
       SomeNonEntityComponent/
+      media/ (shared icons)
+        [no nested directories]
+        icon.svg
   pages/
     bankAccount/
       BankAccountList/
+        [no nested directories]
+
         BankAccountList.page.ts
         BankAccountList.page.test.ts
+        [components for the page should be placed in components folder]
+        [other files should be placed in queries or services]
   queries/
-    bankAccount.query.ts
+    bankAccount.query.ts (query hook)
   services/
     [can be splitted into several files]
-    bankAccount/
+    bankAccount/ (utils, helpers, all the things that should not be put in the components folder)
+      [no nested directories]
       bankAccount.service.ts
       bankAccountJournal.service.ts
+      bankAccount.hook.ts (entity hook)
```

Comment:
it's weird seeing hooks in the `services` folder
even file extension `*.hook.ts` looks strange here
what is the problem to create `hooks` top-level directory and place all `.hooks.ts` files there?

Reply:
The main idea is to keep a structure as simple as possible. Very often hook is a query, and in this case, the folder queries is not needed at all.

Reply:
thx, got it
but according to this logic, we should have `*.query.ts` files inside `services` as well because they are also hooks

Reply:
my guess `*.hook.ts` files should be placed in the `hooks` directory, `*.query.ts` - in the `queries` directory etc
or we should have only `services` directory and place all not-component files there
and differ their purpose by an extension (`hook`, `helper`, `query`, `context` etc)

Reply:
can we just put hooks in `.service.ts` file?
it's unobvious to see smth besides `.service` file here

Reply:
> it's weird seeing hooks in the `services` folder even file extension `*.hook.ts` looks strange here what is the problem to create `hooks` top-level directory and place all `*.hook.ts` files there?
I wrote BankAccountTable.hook.ts just to emphasize is a file with hook inside), I will rename it